### PR TITLE
Updating all instances of now-deprecated `pnpx` to `pnpm dlx`

### LINKED ISSUE #32508: pnpx is deprecated in favor of using `pnpm dlx` and `pnpm exec`
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I'm classifying this as a bug as it's personally a blocker in my dev environment. The cli tool `pnpx` was deprecated in version `6.x` of pnpm, [and removed ](https://github.com/pnpm/pnpm/pull/3652)in `7.x`:

image

I believe they've implemented an alias for retain functionality for this in the near-term but for whatever reason the alias hasn't been working on my local. I've added it manually, but it's still a blocker during the following build step:
`pnpm nx build woocommerce-legacy-assets`

Which is traced back to [a few lines in this file](https://github.com/woocommerce/woocommerce/blob/73409592b7c5b5f24e4bead4d26e5857932e9709/plugins/woocommerce/legacy/project.json#L12). It seems even my manual aliases aren't working while resolving these commands. 
 

### Expected behavior

Build scripts utilize `pnpm exec` and `pnpm dlx` instead of `pnpx`.

### Actual behavior

Various references to `pnpx` within the build scripts, including in the [aforementioned file ](https://github.com/woocommerce/woocommerce/blob/73409592b7c5b5f24e4bead4d26e5857932e9709/plugins/woocommerce/legacy/project.json#L12)which has been a blocker in my environment.

### Steps to reproduce

I'm uncertain of why this has only impacted me thus far (potentially since I'm on Linux), but if the `pnpx` alias is not working (or perhaps if using v7 of pnpm) then the following happens:

1. Run `pnpm nx build woocommerce`
2. Observe the following error:

image

### WordPress Environment

The WooCommerce plugin can't be build, so no system status report, but this is the latest `trunk` with WP 5.9.3. 

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The usage of `pnpx` has a utility has been deprecated in favor of using `pnpm dlx` and `pnpm exec`, as shown by [this issue](https://github.com/pnpm/pnpm/pull/3652) and [the documentation](https://pnpm.io/6.x/pnpx-cli). This PR is intended to update all references. 

_Note: Auto formatting made some broader changes to whitespace in a few of the files. They should align with configured rules, but I can revert those changes if need be as well_

Closes #32508 .

### How to test the changes in this Pull Request:

1. Checkout branch
2. Run `pnpm install && pnpm nx composer-install woocommerce` 
3. Run `pnpm nx build woocommerce`
4. No errors should be encountered, and all just build successfully.
5. Might be best to smoke test other build processes touched by the changes.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/legacy/project.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
 			"options": {
 				"commands": [
 					{
-						"command": "pnpx grunt assets",
+						"command": "pnpm dlx grunt assets",
```
Comment (louwie17) [inline plugins/woocommerce/legacy/project.json]: Should these be using `exec` as well, given `grunt` is being installed by the package.json?

**Review by louwie17 — DISMISSED** [review]
Thanks for working on this @joelclimbsthings 💯 This tested well!
I did leave one comment, but I wasn't entire sure on that, not a blocker for me.

File: plugins/woocommerce/legacy/project.json (hunk 1 above)
Comment (joelclimbsthings) [reply plugins/woocommerce/legacy/project.json]: Yup, that actually still works and should be more efficient. Done in https://github.com/woocommerce/woocommerce/pull/32528/commits/40c6ae32ff13e278492420c20c668fe6d9e25837. Good call 👍🏻

**Comment by joelclimbsthings** [comment]
Thanks @louwie17 , was able to make that change, and also resolved a merge conflict.

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 thanks for updating the one command @joelclimbsthings

**Pull request merged**